src/Runner.Worker/ActionRunner.cs

```diff
@@ -285,25 +284,59 @@ public async Task RunAsync()
 
         }
 
-        public bool TryEvaluateDisplayName(DictionaryContextData contextData, IExecutionContext context)
+        /// <summary>
+        /// Attempts to update the DisplayName.
+        /// As the "Try..." name implies, this method should never throw an exception.
+        /// Returns true if the DisplayName is already present or it was successfully updated.
+        /// </summary>
+        public bool TryUpdateDisplayName(out bool updated)
+        {
+            updated = false;
+
+            // REVIEW:  This try/catch can be removed if some future implementation of EvaluateDisplayName and UpdateTimelineRecordDisplayName
+            // can make reasonable guarantees that they won't throw an exception.
+            try
+            {
+                if (this.Stage == ActionRunStage.Main && EvaluateDisplayName(this.ExecutionContext.ExpressionValues, this.ExecutionContext, out updated))
```

Comment:
Add a comment to clarify why we only attempt this on `Stage == ActionRunStage.Main`

Reply:
Its been a while but I believe a job consists of the following stages

- Job started
  - We try to evaluate all display names here
- PreJobSteps
   - No contexts have changed from the above, so no need to retry if we couldn't fully expand
- Main Steps
    - Context may now have changed, lets update the display names before we start the steps  
- Post Job Steps
    - Name should have already been fully expanded, no need to try again.
- Job complete
    - Nothing to do, job is done don't evaluate display names

Reply:
Thank you!